.github/workflows/functional-baremetal.yaml

```
@@ -0,0 +1,93 @@
+name: functional-baremetal
+on:
+  pull_request:
+    paths:
+      - '^.*baremetal.*$'
+      - '.github/workflows/functional-baremetal.yaml'
+  schedule:
+    - cron: '0 0 * * *'
+jobs:
+  functional-baremetal:
+    strategy:
+      fail-fast: false
+      matrix:
+        name: ["master"]
+        openstack_version: ["master"]
+        ubuntu_version: ["20.04"]
+        include:
+          - name: "xena"
+            openstack_version: "stable/xena"
+            ubuntu_version: "20.04"
+          - name: "wallaby"
+            openstack_version: "stable/wallaby"
+            ubuntu_version: "20.04"
+          - name: "victoria"
+            openstack_version: "stable/victoria"
+            ubuntu_version: "20.04"
+          - name: "ussuri"
+            openstack_version: "stable/ussuri"
+            ubuntu_version: "18.04"
+          - name: "train"
+            openstack_version: "stable/train"
+            ubuntu_version: "18.04"
+    runs-on: ubuntu-${{ matrix.ubuntu_version }}
+    name: Deploy OpenStack ${{ matrix.name }} with Ironic and run baremetal acceptance tests
+    steps:
+      - name: Checkout Gophercloud
+        uses: actions/checkout@v2
+      - name: Deploy devstack
+        uses: EmilienM/devstack-action@v0.6
+        with:
+          branch: ${{ matrix.openstack_version }}
+          conf_overrides: |
+            enable_plugin ironic https://opendev.org/openstack/ironic ${{ matrix.openstack_version }}
+            LIBS_FROM_GIT=pyghmi,virtualbmc
+            FORCE_CONFIG_DRIVE=True
+            Q_AGENT=openvswitch
+            Q_ML2_TENANT_NETWORK_TYPE=vxlan
+            Q_ML2_PLUGIN_MECHANISM_DRIVERS=openvswitch
+            DEFAULT_INSTANCE_TYPE=baremetal
+            OVERRIDE_PUBLIC_BRIDGE_MTU=1400
+            VIRT_DRIVER=ironic
+            BUILD_TIMEOUT=1800
+            SERVICE_TIMEOUT=90
+            GLANCE_LIMIT_IMAGE_SIZE_TOTAL=5000
+            Q_USE_SECGROUP=False
+            API_WORKERS=1
+            IRONIC_BAREMETAL_BASIC_OPS=True
+            IRONIC_BUILD_DEPLOY_RAMDISK=False
+            IRONIC_AUTOMATED_CLEAN_ENABLED=False
+            IRONIC_CALLBACK_TIMEOUT=600
+            IRONIC_DEPLOY_DRIVER=ipmi
+            IRONIC_INSPECTOR_BUILD_RAMDISK=False
+            IRONIC_RAMDISK_TYPE=tinyipa
+            IRONIC_TEMPEST_BUILD_TIMEOUT=720
+            IRONIC_TEMPEST_WHOLE_DISK_IMAGE=False
+            IRONIC_VM_COUNT=1
+            IRONIC_VM_EPHEMERAL_DISK=1
+            IRONIC_VM_LOG_DIR=/opt/stack/new/ironic-bm-logs
+            IRONIC_VM_SPECS_RAM=1024
+            IRONIC_DEFAULT_DEPLOY_INTERFACE=direct
+            IRONIC_ENABLED_DEPLOY_INTERFACES=direct
+            SWIFT_ENABLE_TEMPURLS=True
+            SWIFT_TEMPURL_KEY=secretkey
+          enabled_services: 'ir-api,ir-cond,s-account,s-container,s-object,s-proxy,q-svc,q-agt,q-dhcp,q-l3,q-metai,-cinder,-c-sch,-c-api,-c-vol,-c-bak,-ovn,-ovn-controller,-ovn-northd,-q-ovn-metadata-agent'
```

Interesting, I didn't know you could disable services prefixing them with a `-`. There's a typo though:
```suggestion
          enabled_services: 'ir-api,ir-cond,s-account,s-container,s-object,s-proxy,q-svc,q-agt,q-dhcp,q-l3,q-meta,-cinder,-c-sch,-c-api,-c-vol,-c-bak,-ovn,-ovn-controller,-ovn-northd,-q-ovn-metadata-agent'
```